src/programs/sum_all_eer_files/sum_all_eer_files.cpp

```diff
@@ -0,0 +1,215 @@
+    all_files.Sort( );
+
+    EerFile* current_input_file;
+
+    Image output_sum_image;
+    Image output_dark_image;
+    Image output_gain_image;
+
+    // global and local for threading...
+
+    double* global_sum_image;
+    double* global_sum_squares_image;
+
+    double* sum_image;
+    double* sum_squares_image;
+
+    // find all the Eer files in the current directory..
+
+    wxPrintf("\nThere are %li eer files in this directory.\n", all_files.GetCount( ));
+
+    current_input_file = new EerFile();
+    current_input_file->OpenFile(all_files.Item(0).ToStdString( ), false, false, false, eer_super_res_factor, eer_frames_per_image);
+
+    file_x_size = current_input_file->ReturnXSize( );
+    file_y_size = current_input_file->ReturnYSize( );
+
+    wxPrintf("\nFirst file is %s\nIt is %ix%i sized - all images had better be this size!\n\n", all_files.Item(0), current_input_file->ReturnXSize( ), current_input_file->ReturnYSize( ));
+
+    delete current_input_file;
+
+    output_sum_image.Allocate(file_x_size, file_y_size, 1);
+    output_sum_image.SetToConstant(0.0);
+
+    if ( make_dark_and_gain == true ) {
+        output_dark_image.Allocate(file_x_size, file_y_size, 1);
+        output_gain_image.Allocate(file_x_size, file_y_size, 1);
+    }
+
+    global_sum_image = new double[output_sum_image.real_memory_allocated];
+    ZeroDoubleArray(global_sum_image, output_sum_image.real_memory_allocated);
+
+    if ( make_dark_and_gain == true ) {
+        global_sum_squares_image = new double[output_sum_image.real_memory_allocated];
+        ZeroDoubleArray(global_sum_squares_image, output_sum_image.real_memory_allocated);
+    }
+
+    // loop over all files, and do summing..
+
+    wxPrintf("Summing All Files...\n\n");
+    ProgressBar* my_progress = new ProgressBar(all_files.GetCount( ));
+
+    int number_processed = 0;
+    // thread if available
+#pragma omp parallel default(none) num_threads(max_threads) shared(make_dark_and_gain, global_sum_image, global_sum_squares_image, output_sum_image, all_files, total_summed, number_processed, my_progress, eer_super_res_factor, eer_frames_per_image) private(file_counter, current_input_file, frame_counter, sum_image, sum_squares_image, pixel_counter)
+    { // bracket for omp
+
+        Image buffer_image;
+        int   my_total_summed = 0;
+        sum_image             = new double[output_sum_image.real_memory_allocated];
+        ZeroDoubleArray(sum_image, output_sum_image.real_memory_allocated);
+
+        if ( make_dark_and_gain == true ) {
+            sum_squares_image = new double[output_sum_image.real_memory_allocated];
+            ZeroDoubleArray(sum_squares_image, output_sum_image.real_memory_allocated);
+        }
+
+#pragma omp for
+        for ( file_counter = 0; file_counter < all_files.GetCount( ); file_counter++ ) {
+            //wxPrintf("Summing file %s...\n", all_files.Item(file_counter));
+        	wxPrintf("Summing File %ld\n", file_counter);
```

Comment:
You probably don't mean to have a print statement active in a parallel block, do you?

Reply:
What does active in a parallel block mean? Also no, I was attempting to use this to find which file was corrupted so I'll comment it out

Reply:
What I mean is:
- active : this appears to be a debugging statement that is currently running in all builds, debug or production
- parallel block : the scope that the stastment is in is inside the #pragma omp for at line 136, which means every thread entering that scope will be accessing stdout and printing to the terminal, which can be both confusing and slow down a program a surprising amount.

Reply:
*Sorry, the omp for only divides up work in a parallel way, the parallel region is bounded by the preceding omp clause.